internal/store/hpa.go

```diff
@@ -143,6 +143,56 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_hpa_status_average_value",
+			Type: metric.Gauge,
+			Help: "Average metric value observed by the autoscaler.",
+			GenerateFunc: wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				ms := make([]*metric.Metric, len(a.Status.CurrentMetrics))
+				for i, c := range a.Status.CurrentMetrics {
+					var value int64
+					if c.Type == "Resource" {
```

Comment:
Nit: can we use the constants of `MetricSourceType` -> e.g. `ObjectMetricSourceType `, `PodsMetricSourceType `